scipy/stats/tests/test_morestats.py

```diff
@@ -2460,106 +2463,81 @@ class TestCircFuncs:
-                      [357, 9, 8, 358, 4, 356]])
-        M1 = stats.circmean(x, high=360)
-        M2 = stats.circmean(x.ravel(), high=360)
-        assert_allclose(M1, M2, rtol=1e-14)
-
-        M1 = stats.circmean(x, high=360, axis=1)
-        M2 = [stats.circmean(x[i], high=360) for i in range(x.shape[0])]
-        assert_allclose(M1, M2, rtol=1e-14)
-
-        M1 = stats.circmean(x, high=360, axis=0)
-        M2 = [stats.circmean(x[:, i], high=360) for i in range(x.shape[1])]
-        assert_allclose(M1, M2, rtol=1e-14)
-
-    def test_circvar_axis(self):
-        x = np.array([[355, 5, 2, 359, 10, 350],
-                      [351, 7, 4, 352, 9, 349],
-                      [357, 9, 8, 358, 4, 356]])
-
-        V1 = stats.circvar(x, high=360)
-        V2 = stats.circvar(x.ravel(), high=360)
-        assert_allclose(V1, V2, rtol=1e-11)
-
-        V1 = stats.circvar(x, high=360, axis=1)
-        V2 = [stats.circvar(x[i], high=360) for i in range(x.shape[0])]
-        assert_allclose(V1, V2, rtol=1e-11)
-
-        V1 = stats.circvar(x, high=360, axis=0)
-        V2 = [stats.circvar(x[:, i], high=360) for i in range(x.shape[1])]
-        assert_allclose(V1, V2, rtol=1e-11)
-
-    def test_circstd_axis(self):
-        x = np.array([[355, 5, 2, 359, 10, 350],
-                      [351, 7, 4, 352, 9, 349],
-                      [357, 9, 8, 358, 4, 356]])
-
-        S1 = stats.circstd(x, high=360)
-        S2 = stats.circstd(x.ravel(), high=360)
-        assert_allclose(S1, S2, rtol=1e-11)
-
-        S1 = stats.circstd(x, high=360, axis=1)
-        S2 = [stats.circstd(x[i], high=360) for i in range(x.shape[0])]
-        assert_allclose(S1, S2, rtol=1e-11)
-
-        S1 = stats.circstd(x, high=360, axis=0)
-        S2 = [stats.circstd(x[:, i], high=360) for i in range(x.shape[1])]
-        assert_allclose(S1, S2, rtol=1e-11)
+        x = np.asarray([0.12675364631578953] * 10 + [0.12675365920187928] * 100)
+        circstat = test_func(xp.asarray(x))
+        normal = xp.asarray(numpy_func(x))
+        xp_assert_close(circstat, normal, atol=2e-8)
+
+    @pytest.mark.parametrize('circfunc', [stats.circmean,
+                                          stats.circvar,
+                                          stats.circstd])
+    def test_circmean_axis(self, xp, circfunc):
+        x = xp.asarray([[355, 5, 2, 359, 10, 350],
+                        [351, 7, 4, 352, 9, 349],
+                        [357, 9, 8, 358, 4, 356.]])
+        res = circfunc(x, high=360)
+        ref = circfunc(xp.reshape(x, (-1,)), high=360)
+        xp_assert_close(res, xp.asarray(ref), rtol=1e-14)
+
+        res = circfunc(x, high=360, axis=1)
+        ref = [circfunc(x[i, :], high=360) for i in range(x.shape[0])]
+        xp_assert_close(res, xp.asarray(ref), rtol=1e-14)
+
+        res = circfunc(x, high=360, axis=0)
+        ref = [circfunc(x[:, i], high=360) for i in range(x.shape[1])]
+        xp_assert_close(res, xp.asarray(ref), rtol=1e-12)
```

Comment:
<details>
<summary> local failure with torch + GPU </summary>

```
scipy/stats/tests/test_morestats.py:2518: in test_circmean_axis
    xp_assert_close(res, xp.asarray(ref), rtol=1e-12)
E   AssertionError: Tensor-likes are not close!
E   
E   Mismatched elements: 1 / 6 (16.7%)
E   Greatest absolute difference: 9.5367431640625e-07 at index (4,) (up to 0 allowed)
E   Greatest relative difference: 1.2438287910754298e-07 at index (4,) (up to 1e-12 allowed)
```

</details>

Needed `1.25e-7` for `rtol` locally

Reply:
`6.3e-5` for `var`

Reply:
I think the intent of the tight tolerance here was that the result "should" be exactly the same, since the only difference is that once is explicitly looping over the slices while the other is vectorized. But we know that things just don't work like that, so I suppose we'll resort to default tolerance.